### PR TITLE
api: remove {page,frame,handle}.type() in favor of .fill()

### PR DESCRIPTION
**Pull request by dgozman**
Also introduces `clear` option on `fill` to let user add to existing value.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -2410,13 +2397,36 @@ Bear in mind that if the first click of the `dblclick()` triggers a navigation e
 
 > **NOTE** `elementHandle.dblclick()` dispatches two `click` events and a single `dblclick` event.
 
-#### elementHandle.fill(value)
+#### elementHandle.fill(value[, options])
 - `value` <[string]> Value to set for the `<input>`, `<textarea>` or `[contenteditable]` element.
+- `options` <[Object]>
+  - `clear` <[boolean]> Whether to clear existing value before typing the new one. Defaults to `true`.
```
Comment (JoelEinbinder) [inline docs/api.md]: I think that fill should always clear. Not clearing the input and filling it should almost always be typing. And if they really need it, they can call sendCharacters themselves.

**Comment by pavelfeldman** [comment]
Why are we removing `type`? Because we want users to prefer `fill`?

**Comment by dgozman** [comment]
> Why are we removing `type`? Because we want users to prefer `fill`?

Having both is confusing - it's unclear which one to use. Between type and fill we usually prefer fill.

**Comment by pavelfeldman** [comment]
My only hesitation is discoverability - users will look for `type` and will not find it. They won't be looking for `fill` because the word is too far from the notion of typing on the keyboard. This concerns me to the extent of leaving only `type` with the default semantics of `fill`.

**Comment by jperl** [comment]
I like the idea of type with replace as an option, which defaults to false.

**Pull request closed without merging**